Code/FooBar.php

```diff
@@ -31,4 +31,30 @@ public function checker($number)
 
         return $result;
     }
+
+    public function checkerWithAppend($number)
+    {
+        $result = $this->checker($number);
+
+        if (preg_match('~[0-9]~', $result)) {
+            $result = "";
```

Comment:
I know that this part of the code is kind of unnecessary, but otherwise the result is messed up, if it doesn't divide by 3, 5 or 7. I wanted to minimize the quantity of the code by not copying 90% of first function and therefore just call it.  For me these are two ways of doing that, and none of them seems great for me.